apps/angular/src/core/services/validation.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@angular/core';
+import { AbstractControl, FormGroup, ValidationErrors } from '@angular/forms';
+
+/** 'hasError' function data. */
+type HasErrorData = {
+
+	/** Form. */
+	readonly form: FormGroup;
```

Comment:
```suggestion
	readonly form: FormGroup<T>;
```

let's add a form type